apps/main/src/lend/hooks/useBorrowPositionDetails.ts

```diff
@@ -65,32 +68,39 @@ export const useBorrowPositionDetails = ({
     const thirtyDaysAgo = new Date()
     thirtyDaysAgo.setDate(thirtyDaysAgo.getDate() - 30)
 
-    const recentSnapshots = lendSnapshots.filter((snapshot) => snapshot.timestamp > thirtyDaysAgo)
+    const recentSnapshots = lendSnapshots.filter((snapshot) => new Date(snapshot.timestamp) > thirtyDaysAgo)
 
     if (recentSnapshots.length === 0) return null
 
     return meanBy(recentSnapshots, ({ borrowApy }) => borrowApy) * 100
   }, [lendSnapshots])
 
-  const borrowApy = onChainRatesData?.rates?.borrowApy ?? marketRate?.rates?.borrowApy
-
+  const borrowApy = onChainRatesData?.rates?.borrowApy ?? marketRate?.rates?.borrowApy ?? null
   const collateralTotalValue = useMemo(() => {
     if (!collateralUsdRate || !collateral || !borrowed) return null
     return Number(collateral) * Number(collateralUsdRate) + Number(borrowed)
   }, [collateral, borrowed, collateralUsdRate])
+  const campaignRewards = useMemo(() => {
+    if (!campaigns || !controller) return []
+    return [...(campaigns[controller.toLowerCase()] ?? [])]
+  }, [campaigns, controller])
 
   return {
+    marketType: 'lend',
     liquidationAlert: {
       softLiquidation: status?.colorKey === 'soft_liquidation',
       hardLiquidation: status?.colorKey === 'hard_liquidation',
     },
     health: {
-      value: Number(healthFull),
+      value: healthFull ? Number(healthFull) : null,
       loading: !market || isUserLoanDetailsLoading,
     },
     borrowAPY: {
-      value: borrowApy != null ? Number(borrowApy) : null,
-      thirtyDayAvgRate: thirtyDayAvgRate,
+      rate: borrowApy != null ? Number(borrowApy) : null,
```

Comment:
Double-negatives are hard to read
```suggestion
      rate: borrowApy == null ? null : Number(borrowApy),
```